torch/_C/__init__.pyi.in

```diff
@@ -2101,13 +2101,13 @@ class _CudaEventBase:
         enable_timing: _bool = False,
         blocking: _bool = False,
         interprocess: _bool = False,
-    ) -> _CudaEventBase: ...
+    ) -> Self: ...
     @classmethod
-    def from_ipc_handle(cls, device: _device, ipc_handle: bytes) -> _CudaEventBase: ...
-    def record(self, stream: _CudaStreamBase) -> None: ...
-    def wait(self, stream: _CudaStreamBase) -> None: ...
+    def from_ipc_handle(cls, device: _device, ipc_handle: bytes) -> Self: ...
+    def record(self, stream: Self) -> None: ...
```

Comment:
Here and many below, the type here could be another subclass of Stream and be ok.

Reply:
Self can handle subtype, as given in the examples from [PEP 673](https://peps.python.org/pep-0673/)

Reply:
But here non subclass would be ok?
Like if you have two different subclasses of Stream and do a call on one and use the other as arg. Other will not be the same or subclass of self, yet this call is valid as any Stream subclass would be ok here ?